src/helpers/actions.ts

```diff
@@ -52,3 +52,72 @@ export async function markSpaceAsDeleted(space: string) {
 
   await db.queryAsync(query, [1, space]);
 }
+
+export function refreshProposalsCount(space_ids?: string[]) {
+  return db.queryAsync(
+    `
+      INSERT INTO user_space_activities (proposals_count, user_id, space_id)
+        (SELECT * FROM (
+          SELECT COUNT(proposals.id) AS proposals_count, author, space
+          FROM proposals
+          JOIN spaces ON spaces.id = proposals.space
+          WHERE spaces.deleted = 0
+          ${space_ids ? ' AND space IN (?)' : ''}
+          GROUP BY author, space
+        ) AS t)
+      ON DUPLICATE KEY UPDATE proposals_count = t.proposals_count
+    `.replace(/\n/gm, ' '),
+    space_ids
+  );
+}
+
+export function refreshVotesCount(space_ids: string[]) {
+  return db.queryAsync(
+    `
+      INSERT INTO user_space_activities (votes_count, user_id, space_id)
+        (SELECT * FROM (
+          SELECT COUNT(votes.id) AS votes_count, voter, space
+          FROM votes
+          JOIN spaces ON spaces.id = votes.space
+          WHERE spaces.deleted = 0 AND space IN (?)
+          GROUP BY voter, space
+        ) AS t)
+      ON DUPLICATE KEY UPDATE votes_count = t.votes_count
+    `.replace(/\n/gm, ' '),
+    space_ids
+  );
+}
+
+export function incrementVotesCount(space_id: string, user_id: string) {
+  return db.queryAsync(
+    `
+      INSERT INTO user_space_activities (space_id, user_id, votes_count)
+      VALUES(?, ?, 1)
+      ON DUPLICATE KEY UPDATE votes_count = votes_count + 1
+    `.replace(/\n/gm, ' '),
+    [space_id, user_id]
+  );
+}
+
+export function incrementProposalsCount(space_id: string, user_id: string) {
+  return db.queryAsync(
+    `
+      INSERT INTO user_space_activities (space_id, user_id, proposals_count)
+      VALUES(?, ?, 1)
+      ON DUPLICATE KEY UPDATE proposals_count = proposals_count + 1
+    `.replace(/\n/gm, ' '),
```

Comment:
What is this for?

Reply:
The query is split on multiple line. This end up inserting a new line character `\n` on each new line, which make the SQL query invalid. The `replace()` will remove those invalid new lines characters.

Alternatively, we could also use simple string with simple/double quote, and use concatenation like

```js
'INSERT INTO user_space_activities (space_id, user_id, proposals_count)' +
'VALUES(?, ?, 1)' +
'ON DUPLICATE KEY UPDATE proposals_count = proposals_count + 1'
```

The concatenation solution make copy/paste more difficult, when trying to debug SQL query.

Or have the same query on the same line like this

```js
INSERT INTO user_space_activities (space_id, user_id, proposals_count VALUES(?, ?, 1) ON DUPLICATE KEY UPDATE proposals_count = proposals_count + 1
```

This reduce readability on complex queries.

Will be happy to migrate to any of the alternatives.

Reply:
Removed